### PR TITLE
Traverse and compare json nodes in agnostic mode.

### PR DESCRIPTION
**Pull request by greyerof**
The node's sections CNI, CSI and Hardware are compared in agnostic mode by traversing the unmarshaled json input over a dynamic interface{} type.

The output will appear like this:
```
CNIs: Differences
FIELD                                      CLAIM 1       CLAIM 2
/clus0-0/0/cniVersion                      1.0.1         1.0.0
/clus0-0/0/plugins/0/ipMasq                false         true
/clus0-1/0/plugins/0/hairpinMode           true          false
/clus0-1/0/plugins/0/ipam/routes/0/dst     0.0.0.0/0     0.0.0.1/0

CNIs: Only in CLAIM 1
/clus0-2/0/plugins/0/hairpinMode=true

CNIs: Only in CLAIM 2
<none>

CSIs: Differences
FIELD                         CLAIM 1      CLAIM 2
/metadata/resourceVersion     12734847     12275762

CSIs: Only in CLAIM 1
<none>

CSIs: Only in CLAIM 2
<none>

Hardware: Differences
FIELD                                      CLAIM 1                       CLAIM 2
/clus0-0/IPconfig/14/addr_info/0/local     fe80::58de:73ff:fee0:1b9a     fe80::ccfc:4eff:fee0:1e44
/clus0-0/IPconfig/14/address               5a:de:73:e0:1b:9a             ce:fc:4e:e0:1e:44
...
```

The widths of FIELD and CLAIM 1 columns is dynamic, giving enough room for the longest field path and value.

### CODE REVIEW

File: cmd/tnf/claim/compare/testcases/testcases.go [hunk 1]
```diff
@@ -147,22 +147,30 @@ func GetDiffReport(resultsClaim1, resultsClaim2 claim.TestSuiteResults) DiffRepo
...
+	str += "\n"
+
+	str += "RESULTS DIFFERENCES\n"
+	str += "-------------------\n"
+	if len(r.TestCases) == 0 {
+		str += "<none>\n"
+		return str
+	}
+
+	// str += "\n"
```
Comment (sebrandon1) [inline cmd/tnf/claim/compare/testcases/testcases.go:169]: Commented code.

File: cmd/tnf/claim/compare/diff/diff_test.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
+package diff
+
+import (
+	"encoding/json"
+	"testing"
+
+	"gotest.tools/v3/assert"
```
Comment (sebrandon1) [inline cmd/tnf/claim/compare/diff/diff_test.go:7]: Is this the right `assert`?  Usually we use the stretchr/testify.

**Review by sebrandon1 — COMMENTED** [review]
Some initial comments

File: cmd/tnf/claim/compare/testcases/testcases.go (hunk 1 above)
Comment (greyerof) [reply cmd/tnf/claim/compare/testcases/testcases.go]: Done.

File: cmd/tnf/claim/compare/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package utils
```
Comment (edcdavid) [inline cmd/tnf/claim/compare/utils/utils.go:1]: This directory seems unused

Comment (greyerof) [reply cmd/tnf/claim/compare/utils/utils.go]: That was a leftover from previous local tests. Removed in https://github.com/test-network-function/cnf-certification-test/pull/1386/commits/ea85a69712d057644d1de6d059567a236e8b0d7f

**Pull request merged**